File: LeetCode/cpp/349.cpp
class Solution {
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2) {
        unordered_set<int> s;
        vector<int> nums;
        for (auto n: nums1)
            if (s.find(n) == s.end())
                s.insert(n);
        for (auto n:nums2) {
            if (s.find(n) != s.end()) {
                nums.push_back(n);
                s.erase(n);
            }
        }
        return nums;
    }
}; 

File: LeetCode/python3/349.py
class Solution:
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        nums = []
        s = set(nums1)
        for n in nums2:
            if n in s:
                nums.append(n)
                s.remove(n)
        return nums